src/Google.Api.Gax/Platform.cs

```diff
@@ -112,10 +111,28 @@ public GcePlatformDetails(string metadataJson, string projectId, string instance
         public string InstanceId { get; }
 
         /// <summary>
-        /// The zone name where this GCE instance is running.
+        /// The zone where this GCE instance is running.
+        /// This is in the format <code>projects/&lt;project-number&gt;/zones/&lt;zone-name&gt;</code>.
```

Comment:
When we build an instance of this class we use the TryLoad method which receives the metadata server info, so, as long as the metadata server returns project/121231231/zones/<zone_name> this comment will hold true.
The problem is that the class has a public constructor that clients might be using to pass whatever zone value they want and in that case we cannot obtain a location. Should we:

- Enforce that the value passed is project/234324/zones/<zone_name> (Overkill and more breaking still)
- If we cannot parse zone, then return whatever zone value is in Location?

Reply:
You could always rewrite the comment as "When the details have been retrieved from Google Compute Engine, the value is in the format..."

Reply:
Yes, the comment is not the problem, it was added by me so of course we can have it say whatever. The situation described is independent of the comment.